packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.item-reader-path-s3-object.ts

```diff
@@ -0,0 +1,135 @@
+import { ExpectedResult, IApiCall, IntegTest } from '@aws-cdk/integ-tests-alpha';
+import { App, Duration, RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { DistributedMap, Pass, S3ObjectsItemReaderPath, StateMachine } from 'aws-cdk-lib/aws-stepfunctions';
+
+const TEST_NAME = 'S3ObjectsItemReaderPathTest';
+
+class S3ObjectsItemReaderPathTestStack extends Stack {
+  readonly bucket: Bucket;
+  readonly stateMachine: StateMachine;
+
+  constructor(scope: App, props?: StackProps) {
+    super(scope, `${TEST_NAME}Stack`, props);
+
+    this.bucket = this.createBucket();
+    const distributedMap = this.createDistributedMap();
+    this.stateMachine = this.createStateMachine(distributedMap);
+  }
+
+  private createBucket(): Bucket {
+    return new Bucket(this, `${TEST_NAME}Bucket`, {
+      autoDeleteObjects: true,
+      bucketName: `bucket-item-reader-path-${this.account}-${this.region}`,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+
+  private createDistributedMap(): DistributedMap {
+    const map = new DistributedMap(this, `${TEST_NAME}Map`, {
+      itemReaderPath: new S3ObjectsItemReaderPath({
+        bucketNamePath: '$.bucketName',
+        prefixPath: '$.prefix',
+      }),
+    });
```

Comment:
I'm not too familiar with distributed map, can you explain the series of execution here to help me better understand why we need dynamic bucket name here?

Reply:
Apologies but your question is unclear to me. 

- New construct's aim is allow dynamic bucket name and prefix, hence, they are needed here for tests.
- Test is creating S3 bucket, adding few objects, and letting state machine iterate over them by passing bucket name and prefix as dynamic value.

Rest - the best I can understand question is that it is asking for scenario where dynamic bucket name will be need and for that:

Scenario can be that while creating State Machine, we don't know which S3 bucket will be used for input, hence, we can't statically define it as source for Distributed Map - but we do know that bucket name will be passed inside State Input when State Machine execution is started because assumption is caller knows which S3 bucket to use and caller is responsible for providing proper permissions over S3 bucket for StateMachine (and vice versa).

--- 

Can also refer PR description:
> - For `DistributedMap` state of StepFunctions, `IItemReader` currently only allows S3 bucket as input source to be declared statically in CDK.
> - In other words, current CDK implementation only caters to static use-case where we know either `bucket` or `bucketName` (from which we can create `IBucket`) and pass it to `IItemReader`.
> - Whereas via AWS Console, if we create `DistributedMap` manually, then we can also convey S3 source dynamically using State Input / JsonPath.
> - In other words, for dynamic use-case, we will neither have `bucket` nor `bucketName` i.e. we only know state input variable which will convey `bucketName` e.g. `$.bucketName`.
> 
> ...

Reply:
Ignore the comment. I deployed my own stack and played around with it and now I understood why we need it and how it's used. Appreciate it!

Reply:
Thanks for checking via deployment! 😄